src/test/scala/is/hail/variant/vsm/VSMSuite.scala

```diff
@@ -417,4 +418,31 @@ class VSMSuite extends SparkSuite {
     val vds = hc.importVCF("src/test/resources/sample.vcf.bgz")
     vds.queryGenotypes("gs.map(g => g.gq).hist(0, 100, 100)")
   }
+
+  @Test def testReorderSamples() {
+    val vds = hc.importVCF("src/test/resources/sample.vcf.bgz")
+    val origOrder = Array[Annotation]("C1046::HG02024", "C1046::HG02025", "C1046::HG02026", "C1047::HG00731", "C1047::HG00732")
+    val newOrder = Array[Annotation]("C1046::HG02026", "C1046::HG02024", "C1047::HG00732", "C1046::HG02025", "C1047::HG00731")
+
+    val filteredVds = vds.filterSamplesList(origOrder.toSet)
+    val reorderedVds = filteredVds.reorderSamples(newOrder)
+
+    def getGenotypes(vds: VariantDataset): RDD[((Variant, Annotation), Genotype)] = {
+      val sampleIds = vds.sampleIds
+      vds.rdd.flatMap { case (v, (_, gs)) =>
+        gs.zip(sampleIds).map { case (g, s) =>
+          ((v, s), g)
+        }
+      }
+    }
+
+    assert(getGenotypes(filteredVds).fullOuterJoin(getGenotypes(reorderedVds)).forall { case ((v, s), (g1, g2)) =>
+        g1 == g2
+    })
```

Comment:
Can we also add a test that `reorderSamples(vds.sampleIds)` doesn't change anything?